data-prepper-plugins/translate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/translate/TranslateProcessor.java

```diff
@@ -136,66 +146,124 @@ private void parseFile(String filePath){
+            if (Objects.nonNull(translateProcessorConfig.getTranslateWhen()) && !expressionEvaluator.evaluateConditional(translateProcessorConfig.getTranslateWhen(), recordEvent)) {
                 continue;
             }
             try {
-                String matchKey = record.getData().get(translateProcessorConfig.getSource(), String.class);
-                if(matchesIndividualEntry(record, matchKey) || matchesRangeEntry(record, matchKey) || matchesPatternEntry(record, matchKey)){
-                    continue;
+                if (Objects.nonNull(translateProcessorConfig.getIterateOn())) {
+                    List<Map<String, Object>> nestedObjects = recordEvent.get(translateProcessorConfig.getIterateOn(), List.class);
+                    for (Map<String, Object> nextedObject : nestedObjects) {
+                        performMappings(nextedObject);
+                    }
+                    recordEvent.put(translateProcessorConfig.getIterateOn(), nestedObjects);
+                } else {
+                    performMappings(recordEvent);
                 }
-                else{
+            } catch (Exception ex) {
+                LOG.error(EVENT, "Error mapping the source [{}] of entry [{}]", translateProcessorConfig.getSource(),
+                          record.getData(), ex);
+            }
+        }
+        return records;
+    }
 
-                    // todo : add default, increment metrics, and/or add_tags
+    private String getSourceValue(Object nestedObject, String sourceKey) {
+        if (nestedObject instanceof Map) {
+            return (String) ((Map<?, ?>) nestedObject).get(sourceKey);
+        } else {
+            return ((Event) nestedObject).get(sourceKey, String.class);
+        }
+    }
 
-                }
-            } catch (Exception ex){
-                LOG.error(EVENT, "Error mapping the source [{}] of entry [{}]",
-                        translateProcessorConfig.getSource(), record.getData(), ex);
+    private Object getTargetValue(Object sourceObject, List<String> targetValues){
+        if(sourceObject instanceof String){
+            return targetValues.get(0);
+        }
+        return targetValues;
+    }
+
+    private void performMappings(Object recordObject) {
+        List<String> targetValues = new ArrayList<>();
+        Object sourceObject = translateProcessorConfig.getSource();
+        if (sourceObject instanceof List<?>) {
+            List<String> sourceKeys = (ArrayList<String>) sourceObject;
+            for (String sourceKey : sourceKeys) {
+                String sourceValue = getSourceValue(recordObject, sourceKey);
+                populateTarget(sourceValue, targetValues);
             }
+        } else if (sourceObject instanceof String) {
+            String sourceKey = (String) sourceObject;
+            String sourceValue = getSourceValue(recordObject, sourceKey);
+            populateTarget(sourceValue, targetValues);
         }
```

Comment:
I will add else statement. Although technically, that wont be triggered as we are already checking the type of source in the method isSourceFieldValid() of TranslateProcessorConfig.java and will throw exception if the source option is incorrectly configured. It can only accept a String or List<String>.